typescript/infra/src/core/check.ts

```diff
@@ -32,61 +39,54 @@ export interface ValidatorViolation extends CheckerViolation {
   };
 }
 
-export class AbacusCoreChecker extends AbacusAppChecker<
-  AbacusCore,
-  CoreConfig
+export class AbacusCoreChecker<
+  Networks extends ChainName,
+> extends AbacusAppChecker<
+  Networks,
+  AbacusCore<Networks>,
+  CoreConfig<Networks>
 > {
-  async check(
-    owners: Partial<Record<types.Domain, types.Address>>,
-  ): Promise<void> {
-    await Promise.all(
-      this.app.domainNumbers.map((domain: types.Domain) => {
-        const owner = owners[domain];
-        if (!owner) throw new Error('owner not found');
-        return this.checkDomain(domain, owner);
-      }),
+  async checkOwners(owners: ChainMap<Networks, types.Address>) {
+    return Promise.all(
+      this.app
+        .networks()
+        .map((network) => this.checkDomain(network, owners[network])),
     );
   }
 
-  async checkDomain(domain: types.Domain, owner: types.Address): Promise<void> {
-    await this.checkOwnership(domain, owner);
-    await this.checkProxiedContracts(domain);
-    await this.checkOutbox(domain);
-    await this.checkInboxes(domain);
-    await this.checkAbacusConnectionManager(domain);
-    await this.checkValidatorManagers(domain);
+  async checkDomain(network: Networks, owner: types.Address): Promise<void> {
+    await this.checkDomainOwnership(network, owner);
+    await this.checkProxiedContracts(network);
+    await this.checkOutbox(network);
+    await this.checkInboxes(network);
+    await this.checkAbacusConnectionManager(network);
+    await this.checkValidatorManagers(network);
   }
 
-  async checkOwnership(
-    domain: types.Domain,
+  async checkDomainOwnership(
```

Comment:
I know this is called out elsewhere and maybe out of scope but domain / network terminology here can be confusing

Reply:
https://github.com/abacus-network/abacus-monorepo/issues/296